src/features/upload/UploadStatusLights.jsx

```diff
@@ -50,17 +49,21 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+const formatId = (id) => String(id);
+
 /**
- * Default formatter used to create the row headings when the user specified
- * no formatter.
+ * Returns the row header text for the given items.
+ *
+ * @param {Array} items  The array of items in the row.
+ * @param {function} idFormatter  Function that formats IDs for the row header.
  */
-const defaultRowHeaderFormatter = (_start, _end, items) => {
+const rowHeaderForItems = (items, idFormatter = formatId) => {
```

Comment:
Consider moving this back to `utils/formatting.ts` under a name like `formatIdRange()` just in case we need similar functionality elsewhere. This function looks generic enough that it ideserves its own place in `utils/formatting.ts` (which is the first place I look for a generic formatter function) instead of being tucked away in `src/features/upload/UploadStatusLights.jsx`.

Reply:
Done (not too happy with the name though).